### PR TITLE
[WIP] Studio exit

### PR DESCRIPTION
**Pull request by chrisplatorres**
This PR handles exiting Exokit when user clicks on "Exit" from dropdown menu in Studio.

xitCapture

### CODE REVIEW

File: examples/launcher/src/components/Engine.jsx [hunk 1]
```diff
@@ -106,14 +106,20 @@ class Engine extends React.Component {
...
+
     openMenu(item) {
       const open = this.state.item !== item;
       this.setState({item: open ? item : null}, () => {
         this.postMenuStatus();
       });
     }
-    
+
+    exitExokit() {
```
Comment (chrisplatorres) [inline examples/launcher/src/components/Engine.jsx]: Should this be a different name? `exitStudio()` ?

File: examples/studio.html [hunk 2]
```diff
@@ -905,6 +905,14 @@
           viewport.set(m.data.viewport[0], 1 - m.data.viewport[1] - m.data.viewport[3], m.data.viewport[2], m.data.viewport[3]);
           break;
         }
+        case 'exit': {
+          context.destroy();
+          window.document.resources.addEventListener('drain', () => {
+            console.log('drain');
+            process.exit();
```
Comment (modulesio) [inline examples/studio.html]: `process` should really not be available here. If it is, that is a separate problem.

File: examples/launcher/src/components/Engine.jsx (hunk 1 above)
Comment (modulesio) [reply examples/launcher/src/components/Engine.jsx]: Probably `exitStudio`.

File: examples/studio.html (hunk 2 above)
Comment (chrisplatorres) [reply examples/studio.html]: Oh no, process definitely isn't available. I just noticed I crashed my way to an exit here.

Comment (modulesio) [reply examples/studio.html]: Phew!

**Comment by chrisplatorres** [comment]
Note from Discord:

~25% ? of times studio loads upside down and without UI
25Capture

> it's what I think I mentioned before, the CEF mode being unsupported, since we do in-process it's probably a race condition in the boot it especially triggers if you jiggle the mouse, the first thing to try would be to break it out to the supported multiprocess fork

**Comment by chrisplatorres** [comment]
Going to close this PR to port over to seperate studio repo.

**Comment by modulesio** [comment]
Can we keep it open until that happens, so we don't forget? It's easier for closed PR's to get lost...

**Comment by chrisplatorres** [comment]
Closing now that it is copied in https://github.com/exokitxr/studio/issues/2.

**Pull request closed without merging**